src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -948,37 +985,66 @@ private async Task<IVsWindowFrame> CreateDocWindowForSolutionAsync()
 
         private void ShowManageLibraryPackageForSolutionDialog(object sender, EventArgs e)
         {
+            string parameterString = (e as OleMenuCmdEventArgs)?.InValue as string;
             NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async delegate
             {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                await ShowManageLibraryPackageForSolutionDialogAsync(GetSearchText(parameterString));
+            }).PostOnFailure(nameof(NuGetPackage), nameof(ShowManageLibraryPackageForSolutionDialog));
+        }
 
-                if (ShouldMEFBeInitialized())
+        private async Task ShowManageLibraryPackageForSolutionDialogAsync(string searchText, bool isOpenedByURI = false, string version = null)
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            if (ShouldMEFBeInitialized(isOpenedByURI))
+            {
+                await InitializeMEFAsync();
+            }
+
+            var windowFrame = await FindExistingSolutionWindowFrameAsync();
+            if (windowFrame == null)
+            {
+                //Create the window frame
+                if (isOpenedByURI && !await SolutionManager.Value.IsSolutionAvailableAsync())
                 {
-                    await InitializeMEFAsync();
-                }
+                    EventHandler handler = null;
+                    handler = (s, e) =>
```

Comment:
Please add a comment about why we're hooking up an event handler to run when the solution is finished loading.